src/lib.rs

```diff
@@ -118,6 +121,237 @@ pub fn builder() -> Builder {
     Default::default()
 }
 
+/// Construct a new [`Builder`](./struct.Builder.html) from command line flags.
+pub fn builder_from_flags<I>(args: I) -> Result<Builder, io::Error>
+    where I: Iterator<Item=String>
+{
+    // clap doesn't support the end of options marker, --
+    // https://github.com/kbknapp/clap-rs/issues/735
+    let mut terminated = false;
+    let (bindgen_args, clang_args): (Vec<_>, Vec<_>) = args.partition(|arg| {
+        if terminated { // clang arg
```

Comment:
can we simplify this as: 

```rust
if arg == "--" {
    terminated = true;
}

!terminated
```

Probably renaming terminated to `in_clang_args` or something like that would also be neat.